packages/sdk-client-v3/src/middleware/create-concurrent-modification-middleware.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  Next,
+  Middleware,
+  MiddlewareRequest,
+  MiddlewareResponse,
+} from '../types/types'
+
+export default function createConcurrentModificationMiddleware(): Middleware {
+  return (next: Next) => {
+    return async (request: MiddlewareRequest): Promise<MiddlewareResponse> => {
+      const response = await next(request)
+      if (response.statusCode == 409) {
+        /**
+         * extract the currentVersion
+         * from the error body and update
+         * request with the currentVersion
+         */
+        const version = response.error.body.errors[0].currentVersion
```

Comment:
before executing the request again check that the currentVersion is set. E.g. for deleted resources you get a 409 but no currentVersion

Reply:
Okay, that make sense. However what would then be the value of the version? I am thinking of also checking to ensure the request method is not a 'DELETE' verb.

That is 

```ts
if (response.statusCode == 409 && request.method.toLowerCase() != 'delete') {
   //
}
```

Will this work?

Reply:
No. You will have to do optional chaining:

```
response?.error?.body?.error?.[0]?.currentVersion
```